request_token/decorators.py

```diff
@@ -57,6 +57,10 @@ def inner(*args, **kwargs):
             else:
                 return view_func(*args, **kwargs)
         else:
+            # Check the case when `token` was expired programmatically
+            # using `token.expire()`
+            token.validate_expiration_time()
```

Comment:
Not sure, if it's a good place or better to put it in middleware

Reply:
I would say move it to the middleware - we are already making the db call to fetch the token, so validating the timestamp is no overhead, and it means we fail early, which is good.

Reply:
Done